enhancements/windows-containers/health-management.md

```diff
@@ -0,0 +1,311 @@
+  }
+]
+```
+
+The nodeVariablesToSubsitute and powershellVariablesToSubstitute fields will be used by WICD when processing the
+ConfigMap. They are needed as some Windows services have arguments that require values sourced from either the
+Node object associated with the instance, or data from the instance itself. These fields allow for a ConfigMap that is
+generic enough to apply to all Windows instances being configured, by providing WICD instructions for how to retrieve
+certain values.
+
+For example, if the service ConfigMap had an entry with the following data:
+```
+{
+  "name": "new-service",
+  "command": "C:\new-service --variable-arg1=NODE_NAME --variable-arg2=NETWORK_IP",
+  "nodeVariablesToSubstitute": {
+    "name": "NODE_NAME",
+    "jsonPathNodeObject": "metadata.name"
+  },
+  "powershellVariablesToSubstitute": {
+    "name": "NETWORK_IP",
+    "path": "C:\k\scripts\get_net_ip.ps"
+  },
+  "dependencies": [],
+  "priority": 2
+}
+```
+
+WICD would know to create the service named `new-service` with the value of the --variable-arg1 argument set to the
+name of the instance's Node, and the output of the PowerShell script located at `C:\k\scripts\get_net_ip.ps`, as the
+value of the argument --variable-arg2. WICD will be aware of the proper Node object, as it searching for a Node with an
+internal IP equivalent to the IP of the instance.
+This service has no dependencies, and should be created only after all services with priority 1 have been created.
+
+The `files` key contains the path and checksum of files copied to the instance by WMCO. This will be used to validate
+that essential files have not been tampered with.
+```json
+{
+  "path": "filepath",
+  "checksum": "checksum of file, generated at compile time"
+}
+```
+
+### Node configuration: WMCO responsibilities
+
+1) For a given Machine or BYOH instance, WMCO copies the payload to the instance.
+2) WMCO runs WICD with the bootstrap command, this results in the node object being created.
+3) WMCO annotates the Node object with the `desiredVersion` annotation.
+4) The value of this is set to the current WMCO version.
+5) WMCO starts WICD as a service.
+6) Reconciliation can successfully return at this point.
+
+### Node configuration: WICD responsibilities
+
+For node configuration WICD will have two separate responsibilities:
+
+#### Bootstrap command
+
+When run with the bootstrap command WICD will do the steps required to ensure that Node is created
+for the instance. This functionality is currently implemented within WMCB with the `init-kubelet` command.
+The Node will be cordoned, as the Node is not usable at this point.
+
+#### Running as a Windows service
+
+WICD will have an informer for both ConfigMap and Node events. When it is detected that either the desired version
+annotation has been set for the Node, or the ConfigMap specified by the desired version annotation has been created,
```

Comment:
>When it is detected that either the desired version annotation has been set for the Node

I don't follow why the desired annotation is required. Please elaborate on that.